Sources/SwiftKafka/KafkaConfig.swift

```diff
@@ -0,0 +1,67 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the swift-kafka-gsoc open source project
+//
+// Copyright (c) 2022 Apple Inc. and the swift-kafka-gsoc project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of swift-kafka-gsoc project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Crdkafka
+
+public class KafkaConfig {
```

Comment:
Can we add a doc comment here please. Also pointing to the librdkafka docs.

I am inclined to call this `LIBRDKafkaConfig`

Reply:
IMO, it depends on how we want to expose the config Kafka config to the user — I would stick with the `KafkaConfig` name for the public interface. However, if we want to abstract these key/value pairs e.g. by having a struct with strongly typed configuration properties, we can consider having an `internal` `LIBRDKafka` class for the key/value pair implementation that is then accessed by our `public` `KafkaConfig` struct.

Reply:
I think we need to go the route of having a non-strongly typed struct in the beginning. This leaves the greatest flexibility. 

Furthermore, I think we should slightly change this type. Instead of it being a class we should make it a struct with an internal class. The struct should implement copy-on-write semantics. In the case of a copy we would dupe the pointer. This allows us to not dupe the pointer when we pass it to librdkafka. You probably want to use this: https://developer.apple.com/documentation/swift/isknownuniquelyreferenced(_:)-98zpp